internal/parse/parser.go

```diff
@@ -142,13 +144,38 @@ func (p *Parser) Parse(input string) (expr *ParsedExpr, err error) {
 		}
 
 		// If nothing above can be parsed we advance the parser.
-		p.pos++
+		p.advance()
 	}
 	// Add any remaining unparsed string input to the parser.
 	p.add(nil)
 	return &ParsedExpr{p.parts}, nil
 }
 
+// advance increments p.pos until we reach a space, a $, a " or a '. If we find
+// a space we advance to the rightmost adjacent space so the p.pos points to the
+// last space before the next non-space character.
```

Comment:
Similarly here, the documentation is copying the code in English form, which is also an anti-pattern. Instead, the best comments provide context without repeating the code. For example:
```
// advance moves p.pos forward until it reaches content that may be the start of known tokens
// that must be taken into account by the parser.
```
That will indeed save the next reader the trouble of implying why this function was created, while allowing them to read the code if they would like to know details of that process.

To be clear, I'm nitpicking more than usual here to attempt to establish some shared baselines. I won't be bothering you for too long about this. :)

Reply:
Changed.